internal/webhook/external/quay_test.go

```diff
@@ -0,0 +1,188 @@
+package external
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/indexer"
+)
+
+func TestQuayHandler(t *testing.T) {
+	const testURL = "https://webhooks.kargo.example.com/nonsense"
+
+	const testProjectName = "fake-project"
+
+	testScheme := runtime.NewScheme()
+	require.NoError(t, kargoapi.AddToScheme(testScheme))
+
+	const testToken = "mysupersecrettoken"
+	testSecretData := map[string][]byte{
+		quaySecretDataKey: []byte(testToken),
+	}
+
+	for _, testCase := range []struct {
+		name       string
+		client     client.Client
+		secretData map[string][]byte
```

Comment:
This field isn't relevant for any of these test cases because the handler doesn't do anything with the secret. The secret's only used in url construction and that has no relevance to testing the handler

```suggestion
```